Sources/Views/InputTextView.swift

```diff
@@ -303,26 +303,50 @@ open class InputTextView: UITextView {
     private func parseForComponents() -> [Any] {
         
         var components = [Any]()
-        let range = NSRange(location: 0, length: attributedText.length)
-        attributedText.enumerateAttributes(in: range, options: []) { (object, range, _) in
-            
-            if object.keys.contains(.attachment) {
-                if let attachment = object[.attachment] as? NSTextAttachment {
-                    if let image = attachment.image {
-                        components.append(image)
-                    } else if let image = attachment.image(forBounds: attachment.bounds,
-                                                           textContainer: nil,
-                                                           characterIndex: range.location) {
-                        components.append(image)
+        var attachments = [(NSRange, UIImage)]()
+        let length = attributedText.length
+        let range = NSRange(location: 0, length: length)
+        attributedText.enumerateAttribute(.attachment, in: range) { (object, range, _) in
+            if let attachment = object as? NSTextAttachment {
+                if let image = attachment.image {
+                    attachments.append((range, image))
+                } else if let image = attachment.image(forBounds: attachment.bounds,
+                                                       textContainer: nil,
+                                                       characterIndex: range.location) {
+                    attachments.append((range,image))
+                }
+            }
+        }
+        
+        var curLocation = 0
+        if attachments.count == 0 {
+            let text = attributedText.string.trimmingCharacters(in: .whitespacesAndNewlines)
+            if !text.isEmpty {
+                components.append(text)
+            }
+        }
+        else {
+            attachments.forEach { (attachment) in
+                let (range, image) = attachment
+                if curLocation < range.location {
+                    let textRange = NSMakeRange(curLocation, range.location)
+                    let text = attributedText.attributedSubstring(from: textRange).string.trimmingCharacters(in: .whitespacesAndNewlines)
+                    if !text.isEmpty {
+                        components.append(text)
                     }
                 }
-            } else {
-                let stringValue = attributedText.attributedSubstring(from: range).string.trimmingCharacters(in: .whitespacesAndNewlines)
-                if !stringValue.isEmpty {
-                    components.append(stringValue)
+                
+                curLocation = range.location + range.length
+                components.append(image)
+            }
+            if curLocation < length - 1  {
```

Comment:
After last `image`, maybe still has text, so we should append it to `components`.

Reply:
Why is `length - 1` here? Sorry 😓

Reply:
:smile: It's because we add a new line after image, so we decrease 1 to prevent unnecessary calculation.